translate-c seems to get the types wrong for C code translated to zig based on zig compiler error

### Zig Version

0.11.0-dev.1384+fe6dcdba1

### Steps to Reproduce and Observed Behavior

On a MacBook Air (M1, 2020) with MacOS 12.6.2
```bash
# get micromamba
curl micro.mamba.pm/install.sh | bash

# make new micromamba environment
micromamba create -n cmake -c bioconda -c conda-forge -c pkgs cmake=3.25.1

# activate the micromamba environment
micromamba activate cmake

# determine cmake version
cmake --version

cmake version 3.25.1

# download LLVM & clang 15.0.7 
# had to use browser, curl not working?
cd ~/Downloads/
tar xJf clang+llvm-15.0.7-arm64-apple-darwin22.0.tar.xz

# zig fe6dcdb
cd ~/git
git clone https://github.com/ziglang/zig
cd zig
git checkout fe6dcdb

# build zig with LLVM and clang 15.0.7
# Edit - forgot about build directory steps
rm -fr build
mkdir -p build
cd build
cmake .. -DZIG_STATIC_LLVM=ON -DCMAKE_PREFIX_PATH=/Users/jean.elbers/Downloads/clang+llvm-15.0.7-arm64-apple-darwin22.0/bin
make install

# add zig to path of course
export PATH=/Users/jean.elbers/git/zig/build/stage3/bin:$PATH

# get C file to translate to zig
cd ~/git/test
curl 'https://raw.githubusercontent.com/lh3/biofast/4177ef282f4e1addc5b20beb7ee2f6435806f844/fqcnt/fqcnt_c1_kseq.c' -O

# modify include statement
perl -pi -e "s/#include <zlib.h>/#include \"zlib.h\"/" fqcnt_c1_kseq.c

# get other C library
curl 'https://raw.githubusercontent.com/lh3/biofast/4177ef282f4e1addc5b20beb7ee2f6435806f844/lib/kseq.h' -O

# get zlib.h version 1.2.11 and also zconf.h
cp /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/zlib.h .
cp /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/zconf.h .

# convert c to zig
zig translate-c fqcnt_c1_kseq.c > fqcnt_c1_kseq.zig
```

```bash
# try to compile
zig build-exe fqcnt_c1_kseq.zig

fqcnt_c1_kseq.zig:1822:13: error: expected type 'c_int', found 'bool'
        c = tmp;
            ^~~
referenced by:
    main: fqcnt_c1_kseq.zig:1846:21
    comptime_0: /Users/jean.elbers/git/zig/build/stage3/lib/zig/std/start.zig:59:50
    remaining reference traces hidden; use '-freference-trace' to see all reference traces
```

### Expected Behavior

Compiles without errors.

Doing the above, except using system libraries directly and not modifying the `#include <zlib.h>`
```bash
zig translate-c -lz fqcnt_c1_kseq.c > fqcnt_c1_kseq.zig
zig build-exe -lz fqcnt_c1_kseq.zig

fqcnt_c1_kseq.zig:1869:13: error: expected type 'c_int', found 'bool'
        c = tmp;
            ^~~
referenced by:
    main: fqcnt_c1_kseq.zig:1893:21
    comptime_0: /Users/jean.elbers/git/zig/build/stage3/lib/zig/std/start.zig:59:50
    remaining reference traces hidden; use '-freference-trace' to see all reference traces
```

results in the same expected type `X` found `Y` error

--- a/src/translate_c.zig
+++ b/src/translate_c.zig
@@ -4519,7 +4519,10 @@ fn transCreateNodeAssign(
     defer block_scope.deinit();
 
     const tmp = try block_scope.makeMangledName(c, "tmp");
-    const rhs_node = try transExpr(c, &block_scope.base, rhs, .used);
+    var rhs_node = try transExpr(c, &block_scope.base, rhs, .used);
+    if (!exprIsBooleanType(lhs) and isBoolRes(rhs_node)) {
+        rhs_node = try Tag.bool_to_int.create(c.arena, rhs_node);
+    }
     const tmp_decl = try Tag.var_simple.create(c.arena, .{ .name = tmp, .init = rhs_node });
     try block_scope.statements.append(tmp_decl);
 

--- a/test/translate_c.zig
+++ b/test/translate_c.zig
@@ -3900,4 +3900,20 @@ pub fn addCases(cases: *tests.TranslateCContext) void {
         \\pub const ZERO = @as(c_int, 0);
         \\pub const WORLD = @as(c_int, 0o0000123);
     });
+
+    cases.add("Assign expression from bool to int",
+        \\void foo(void) {
+        \\    int a;
+        \\    if (a = 1 > 0) {}
+        \\}
+    , &[_][]const u8{
+        \\pub export fn foo() void {
+        \\    var a: c_int = undefined;
+        \\    if ((blk: {
+        \\        const tmp = @boolToInt(@as(c_int, 1) > @as(c_int, 0));
+        \\        a = tmp;
+        \\        break :blk tmp;
+        \\    }) != 0) {}
+        \\}
+    });
 }

